job/job.py

```diff
@@ -6,20 +6,66 @@
 import time
 from datetime import datetime, timedelta
 
```

Comment:
🔧 So I don't think we should be changing this file yet at all. In the interest of small, specific PRs, this should add the fetch code that you have added BUT not yet update the actual pipeline. This encourages a few things. 1) It encourages us to validate new code with tests as opposed to running it full-out manually, 2) it makes it easier to trace specific changes (and therefore bugs) to specific commits/PRs instead of a big ol' change that not only adds new code but also refactors things, and 3) there is an argument to be made for organization. If this PR simply adds a tool to the tool belt, then a later PR uses that tool in the pipeline, we simply help our future selves a lot with cleaner, smaller, simpler, more specific changes. Happy to be disagreed with! But I suggest that this file is reverted (barring the import updates due to other refactoring).

Reply:
Sounds good! Sorry for getting ahead of myself!

Reply:
Restored, except for the import updates you mentioned!